### DISCUSSION TITLE
How to create custom cacheKey?

### DISCUSSION THREAD

**Discussion by gijsdh** [MarcGiffing/bucket4j-spring-boot-starter#133] [Q&A] [answered]
So i want to create a custom cacheKey based on a attribute defined in a earlier filter.

So how would I go about that?  
Would below work, or do I need to create a filter?
I am unsure how to get the httpServletRequest in the Bean. 

```
@Service
public class MyService{
      
        @Context 
        HttpServletRequest  httpServletRequest;
      
	public String userId() {
		return Parse(request.getAttribute("id"));
	}
} 

```
expression = @myService.userId()

**Comment by MarcGiffing** [comment, marked as answer]
The root object in the expression is the HttpServletRequest object. It's properties are accessed directly:

`cache-key=getAttribute("id")`


- servlet: jakarta.servlet.http.HttpServletRequest (e.g. getRemoteAddr() or getRequestURI())
- webflux: org.springframework.http.server.reactive.ServerHttpRequest
- gateway: org.springframework.http.server.reactive.ServerHttpRequest
